examples/demo10.yaml

```diff
@@ -100,12 +103,18 @@ gal :
             # However, sequences are indexed by the overall object number, so in order to
             # step through the items in this list once for each time it gets to the "first"
             # item, we need to set the index to repeat 20 times before incrementing.
-            # We again use the convenient YAML variable name notation to do so.
-            index : { type : Sequence , repeat : *num_in_ring }
+            index :
+                type : Sequence
+                # You can access the current value of any other item in the config dict using
+                # type = Current.  The syntax for the key is to treat the config levels as
+                # attributes.  So the number of galaxies in the ring defined above is gal.num.
+                # You can use any value this way.  If it hasn't actually been evaluated yet,
+                # it will evaluate it as needed for the current object.
+                repeat : { type : Current, key : gal.num }
```

Comment:
The YAML version works too.  I just wanted to introduce the "Current" type here for pedagogical reasons, not because the other way doesn't work any more.  

In particular, it seemed useful to have a couple Current types early in this demo before the ones in the truth catalog section.  This and pixel_scale later are both constants, but the ones in the truth output are not.  That's the main advantage of Current over the YAML `*` notation.  The former gets the current value of things that may change from one galaxy to another.  The `*` notation only works for constants within the YAML file.

I also moved the introduction of `eval_variables` to demo11, since its use is a bit richer in that demo than what we had here.  It had been a fairly lame example in this demo, imo.  But combining it with the YAML tags is a nice design pattern that I've used in practice to put all the constants you might want to modify all together near the top of the config file.